src/qcodes/instrument_drivers/mock_instruments/__init__.py

```diff
@@ -939,8 +939,8 @@ def get_raw(self) -> ParamRawDataType:
 
 
 def setpoint_generator(
-    *sp_bases: Sequence[float] | np.ndarray,
-) -> tuple[np.ndarray | Sequence[float], ...]:
+    *sp_bases: Sequence[float | np.floating] | np.ndarray,
```

Comment:
isn't float part of np.floating?
```suggestion
    *sp_bases: Sequence[np.floating] | np.ndarray,
```

Reply:
No these are distinct types (that represents the same type of data)

Reply:
```
>>> a = 3.2
>>> isinstance(a, float)
True
>>> isinstance(a, np.float64)
False
>>> isinstance(a, np.floating)
False
>>>
```